chips/e310x/src/chip.rs

```diff
@@ -144,16 +78,141 @@ impl kernel::Chip for E310x {
     }
 }
 
+pub unsafe fn handle_trap() {
+    let cause = rv32i::riscvregs::mcause::read();
+    // if most sig bit is set, is interrupt
+    if cause.is_interrupt() {
+        // strip off the msb
+        match rv32i::riscvregs::mcause::Interrupt::from(cause.code()) {
+            rv32i::riscvregs::mcause::Interrupt::MachineSoft => {
+                debug!("encountered machine mode software interrupt");
+            }
+            // should never occur
+            rv32i::riscvregs::mcause::Interrupt::UserSoft => (),
+            rv32i::riscvregs::mcause::Interrupt::SupervisorSoft => (),
+
+            rv32i::riscvregs::mcause::Interrupt::MachineTimer => (),
+
+            // should never occur
+            rv32i::riscvregs::mcause::Interrupt::UserTimer => (),
+            rv32i::riscvregs::mcause::Interrupt::SupervisorTimer => (),
+
+            // this includes UART, GPIO pins, etc
+            rv32i::riscvregs::mcause::Interrupt::MachineExternal => {
+                // just send out a message that the interrupt occurred and complete it
+                let ext_interrupt_wrapper = plic::next_pending();
+                match ext_interrupt_wrapper {
+                    None => (),
+                    Some(ext_interrupt_id) => {
+                        debug!("interrupt triggered {}\n", ext_interrupt_id);
+                        plic::complete(ext_interrupt_id);
+                        plic::surpress_all();
+                    }
+                }
+            }
+            // should never occur
+            rv32i::riscvregs::mcause::Interrupt::UserExternal => (),
+            rv32i::riscvregs::mcause::Interrupt::SupervisorExternal => (),
```

Comment:
I'm still getting the hang or Rust, but will this provide enough information to debug if it does occur?

Reply:
I think that would be useful to add some prints in here